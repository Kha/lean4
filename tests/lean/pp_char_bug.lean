#check (fin.mk 2 dec_trivial : fin 5)
#check (fin.mk 1 dec_trivial : fin 3)
#check 'a'
#check to_string 'a'
#eval to_string 'a'
#eval 'a'

#eval char.of_nat 1
#eval char.of_nat 1
#eval char.of_nat 0x10
#eval char.of_nat 0x1000
#eval char.of_nat 0x10ffff

#check char.of_nat 1
#check char.of_nat 20
#check char.of_nat 15
#check char.of_nat 16
#eval char.of_nat 15
#eval char.of_nat 16

example : char.of_nat 1 = '\x01' :=
rfl
example : char.of_nat 20 = '\x14' :=
rfl
