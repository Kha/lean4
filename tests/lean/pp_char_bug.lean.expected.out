⟨2, _⟩ : fin 5
⟨1, _⟩ : fin 3
'a' : char
to_string 'a' : string
"a"
'a'
'\u0001'
'\u0001'
'\u0010'
'\u1000'
'\u10ffff'
'\u0001' : char
'\u0014' : char
'\u000f' : char
'\u0010' : char
'\u000f'
'\u0010'
